plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs},
+				dailyVolumeByDateQuery: accountConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - account and market ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+				dailyVolumeByDateQuery: accountMarketConfigVolumeQuery,
+			},
+		},
+	}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			actual, e := makeFilterVolume(
+				k.configValue,
+				k.exchangeName,
+				k.tradingPair,
+				k.assetDisplayFn,
+				k.baseAsset,
+				k.quoteAsset,
+				k.db,
+				k.config,
+			)
+
+			assert.Equal(t, k.wantError, e)
+			assert.Equal(t, k.wantSubmitFilter, actual)
+		})
+	}
+}
+
+func TestVolumeFilterFn(t *testing.T) {
+	db := connectTestDb()
+	dailyVolumeByDateQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	emptyFilter := &volumeFilter{
+		name:                   "volumeFilter",
+		configValue:            "",
+		baseAsset:              testNativeAsset,
+		quoteAsset:             testNativeAsset,
+		config:                 &VolumeFilterConfig{},
+		dailyVolumeByDateQuery: dailyVolumeByDateQuery,
+	}
+
+	testCases := []struct {
+		name      string
+		filter    *volumeFilter
+		dailyOTB  *VolumeFilterConfig
+		dailyTBB  *VolumeFilterConfig
+		op        *txnbuild.ManageSellOffer
+		wantOp    *txnbuild.ManageSellOffer
+		wantError error
+		wantTBB   *VolumeFilterConfig
+	}{
+		{
+			name:      "failure - is selling",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.CreditAsset{}},
+			wantOp:    nil,
+			wantError: fmt.Errorf("error when running the isSelling check for offer '{Selling:{Code: Issuer:} Buying:{} Amount: Price: OfferID:0 SourceAccount:<nil>}': invalid assets, there are more than 2 distinct assets: sdexBase={native  }, sdexQuote={native  }, selling={ }, buying={}"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "failure - invalid sell price in op",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
```

Comment:
you've got some very comprehensive tests here! 👍 

However, that brings up challenges of it's own. I think we can take more of a whitebox testing approach where we only test the logic that is more likely to have errors, which will keep the tests a little smaller and more targeted, making them easier to reason about.

for example, it's safe to assume that the offer passed in is fully valid and has a valid price and that parse float will not fail, and that we call parsefloat on it correctly.

If we want to test the functionality more thoroughly, the best approach would be to extract the function that does the validation and extraction of values from the various inputs and write a functional test for it. this will ensure we have smaller and more targeted tests. I'm not opposed to doing this, I just don't think it's necessary yet because we have larger gaping holes in our test infrastructure that can benefit from test coverage.